vl53v1x/vl53l1x.go

```diff
@@ -0,0 +1,541 @@
+}
+
+// readResults read the register and stores the data in the results buffer
+func (d *Device) readResults() {
+	data := make([]byte, 17)
+	msb := byte((RESULT_RANGE_STATUS >> 8) & 0xFF)
+	lsb := byte(RESULT_RANGE_STATUS & 0xFF)
+	d.bus.Tx(Address, []byte{msb, lsb}, data)
+	d.results.status = data[0]
+	// data[1] report_status : not used
+	d.results.streamCount = data[2]
+	d.results.effectiveSPADCount = readUint(data[3], data[4])
+	// data[5] , data[6] peak signal count rate mcps sd0 : not used
+	d.results.ambientRateMCPSSD0 = readUint(data[7], data[8])
+	// data[9] , data[10] sigma_sd0 : not used
+	// data[11] , data[12] phase_sd0 : not used
+	d.results.mmCrosstalkSD0 = readUint(data[13], data[14])
+	d.results.signalRateCrosstalkMCPSSD0 = readUint(data[15], data[16])
+}
+
+// dataReady returns true when the data is ready to be read
+func (d *Device) dataReady() bool {
+	return (d.readReg(GPIO_TIO_HV_STATUS) & 0x01) == 0
+}
+
+// Distance returns the distance in mm
+func (d *Device) Distance() int32 {
+	return int32(d.rangingData.mm)
+}
+
+// Status returns the status of the sensor
+func (d *Device) Status() RangeStatus {
+	return d.rangingData.status
+}
+
+// SignalRate returns the peak signal rate in count per second (cps)
+func (d *Device) SignalRate() int32 {
+	return int32(1000000 * d.rangingData.signalRateMCPS)
+}
+
+// AmbientRate returns the ambient rate in count per second (cps)
+func (d *Device) AmbientRate() int32 {
+	return int32(1000000 * d.rangingData.ambientRateMCPS)
+}
+
+// getRangingData stores in the buffer the ranging data
+func (d *Device) getRangingData() {
+	d.rangingData.mm = uint16((uint32(d.results.mmCrosstalkSD0)*2011 + 0x0400) / 0x0800)
+	switch d.results.status {
+	case 17: // MULTCLIPFAIL
+	case 2: // VCSELWATCHDOGTESTFAILURE
+	case 1: // VCSELCONTINUITYTESTFAILURE
+	case 3: // NOVHVVALUEFOUND
+		d.rangingData.status = HardwareFail
+		break
+
+	case 13: // USERROICLIP
+		d.rangingData.status = MinRangeFail
+		break
+
+	case 18: // GPHSTREAMCOUNT0READY
+		d.rangingData.status = SynchronizationInt
+		break
+
+	case 5: // RANGEPHASECHECK
+		d.rangingData.status = OutOfBoundsFail
+		break
+
+	case 4: // MSRCNOTARGET
+		d.rangingData.status = SignalFail
+		break
+
+	case 6: // SIGMATHRESHOLDCHECK
+		d.rangingData.status = SignalFail
+		break
+
+	case 7: // PHASECONSISTENCY
+		d.rangingData.status = WrapTargetFail
+		break
+
+	case 12: // RANGEIGNORETHRESHOLD
+		d.rangingData.status = XtalkSignalFail
+		break
+
+	case 8: // MINCLIP
+		d.rangingData.status = RangeValidMinRangeClipped
+		break
+
+	case 9: // RANGECOMPLETE
+		if d.results.streamCount == 0 {
+			d.rangingData.status = RangeValidNoWrapCheckFail
+		} else {
+			d.rangingData.status = RangeValid
+		}
+		break
+
+	default:
+		d.rangingData.status = None
+	}
+
+	d.rangingData.signalRateMCPS = float32(d.results.signalRateCrosstalkMCPSSD0) / (1 << 7)
+	d.rangingData.ambientRateMCPS = float32(d.results.ambientRateMCPSSD0) / (1 << 7)
```

Comment:
It looks like `signalRateMCPS` and `ambientRateMCPS` are converted to float here but only used as integers, so I think this int -> float -> int conversion can be removed.